oci/spec_opts_linux_test.go

```diff
@@ -106,3 +110,114 @@ func TestDropCaps(t *testing.T) {
 		}
 	}
 }
+
+func TestGetDevices(t *testing.T) {
+	dir, err := ioutil.TempDir("/dev", t.Name())
+	if err != nil {
+		if os.IsPermission(err) {
+			t.Skip("Test requires root")
+		}
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	zero := filepath.Join(dir, "zero")
+	if err := ioutil.WriteFile(zero, nil, 0600); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := unix.Mount("/dev/zero", zero, "", unix.MS_BIND, ""); err != nil {
+		t.Fatal(err)
+	}
+	defer unix.Unmount(filepath.Join(dir, "zero"), unix.MNT_DETACH)
```

Comment:
why use `MNT_DETACH` here?

Reply:
Why not? It ensures the amount is eventually cleaned up if it can't be cleaned up immediately.

Reply:
We also aren't checking the error and even if we were there would be nothing actionable. I think `MNT_DETACH` is an appropriate use here because it ensures the mount cleanup.